samples/cellular/nrf_cloud_rest_device_message/src/main.c

```diff
@@ -351,6 +509,13 @@ static int init(void)
 		return -EFAULT;
 	}
 
+	/* If provisioning library is disabled, ensure device has credentials installed
+	 * before proceeding
+	 */
+	if (!IS_ENABLED(CONFIG_NRF_PROVISIONING)) {
+		await_credentials();
```

Comment:
You could put the await_credentials call in `setup()` after the call to `init_provisioning` and then you would only need that single call instead of two separate pathways for calling `await_credentials`

Reply:
that would look nicer, but functionally i feel it belongs here; no point continuing beyond this point if the creds don't exist.